### PR TITLE
Add experimental "Labs" section to settings

### PR DESCRIPTION
**Pull request by aviraldg**

### CODE REVIEW

File: src/UserSettingsStore.js [hunk 1]
```diff
@@ -112,4 +112,14 @@ module.exports = {
             append: true,  // We always append for email pushers since we don't want to stop other accounts notifying to the same email address
         });
     },
+
+    isFeatureEnabled: function(feature: string): boolean {
+        feature = feature.match(/\w+/g).join('_').toLowerCase();
```
Comment (dbkr) [inline src/UserSettingsStore.js]: Not sure I'm a big fan of the magic here - I'd prefer `LABS_FEATURES` to just be an object.

File: src/components/structures/UserSettings.js [hunk 2]
```diff
@@ -357,6 +362,32 @@ module.exports = React.createClass({
...
+                           onChange={e => UserSettingsStore.setFeatureEnabled(feature, e.target.checked)} />
+                    <label htmlFor={feature}>{feature}</label>
+                </div>
+            ));
+            return (
+                <div>
+                    <h3>Labs</h3>
+
+                    <div className="mx_UserSettings_section">
+                        <p>These experimental features may change, break or disappear at any time. We make absolutely no guarantees about what may happen if you turn one of these experiments on, and your client may even spontaneously combust. Jokes aside, your client may delete all your data or your security and privacy could be compromised in unexpected ways. Please proceed with caution.</p>
```
Comment (dbkr) [inline src/components/structures/UserSettings.js]: Not sure I'd copy the string verbatim, but I'm happy to make up some text to go here.

File: src/components/structures/UserSettings.js [hunk 3]
```diff
@@ -357,6 +362,32 @@ module.exports = React.createClass({
...
+                        {features}
+                        <br />
+                        <div className="mx_UserSettings_button" onClick={() => window.location.reload()}>Restart Vector</div>
```
Comment (dbkr) [inline src/components/structures/UserSettings.js]: Is this necessary? I don't think we need it for the current two, and if future features need to know to update, we can dispatch through the dispatcher.

**Comment by dbkr** [comment]
Other than that, looking good.

File: src/components/structures/UserSettings.js (hunk 2 above)
Comment (aviraldg) [inline src/components/structures/UserSettings.js]: It's actually been modified a bit. Happy to replace it if you have suggestions.

File: src/UserSettingsStore.js (hunk 1 above)
Comment (aviraldg) [inline src/UserSettingsStore.js]: Yes, that's how they're intended to be used -- with underscored strings (which remain the same after this magic.)

**Pull request merged**